js/admin.js

```diff
@@ -32,75 +39,150 @@ jQuery( document ).ready(
-		$( '#lang_list' ).on(
-			'selectmenucreate selectmenuselect',
-			function() {
-				selectmenuRefreshButtonText( this );
+		// Create the jQuery UI selectmenu widget for flag list dropdown and return its instance.
+		// There is no need of create and select callbacks with jQuery UI 1.12 because overrinding _renderButtonItem method do the job.
+		if ( isJqueryUImin112 ) {
+			selectmenuFlagListCallbacks =
+				{
+					open: openCallback,
+				};
+		} else {
+			selectmenuFlagListCallbacks = {
+				create: createSelectCallback,
+				select: createSelectCallback,
+				open: openCallback,
+			};
+		}
+
+		// Create the selectmenu widget only if the field is present.
+		if ( flagListExist ) {
+			var selectmenuFlagList = $( '#flag_list' ).selectmenu( Object.assign( {}, selectmenuOptions, selectmenuFlagListCallbacks ) ).selectmenu( 'instance' );
+			// Overrides each item in the jQuery UI selectmenu list by injecting flag image.
+			selectmenuFlagList._renderItem = selectmenuRenderItem;
+			// Override the selected item rendering for jQuery UI 1.12
+			if ( isJqueryUImin112 ) {
+				selectmenuFlagList._renderButtonItem = selectmenuRenderButtonItem;
+				selectmenuFlagList.refresh(); // Need to refresh to take in account the button item rendering method after the selectmenu widget instanciaion.
 			}
-		);
-		// Create the jQuery UI selectmenu widget
-		$( '#lang_list' ).selectmenu( { width: '97%' } ); // jQuery UI selectmenu widget substract 2% and we need 95% for the width matches to the other fields width.
-		// However for the wizard we need a 100% width.
-		if( $( '#lang_list' ).closest( '.pll-wizard-content' ).length > 0 ) {
-			$( '#lang_list' ).selectmenu( 'option', 'width', '102%' );
 		}
-		// Overrides each element in the jQuery UI selectmenu list by injecting flag image.
-		$( '#lang_list' ).selectmenu( 'instance' )._renderItem = selectmenuRenderItem;
+		// Language choice in predefined languages in Polylang Languages settings page and wizard.
+		// Overrides the predefined language dropdown list with our customized jQuery ui selectmenu widget.
 
-		// Languages form
-		// Fills the fields based on the language dropdown list choice
-		$( '#add-lang #lang_list' ).on(
-			'selectmenuchange',
-			function() {
-				var value = $( this ).val().split( ':' );
-				var selected = $( "option:selected", this ).text().split( ' - ' );
-				$( '#lang_slug' ).val( value[0] );
-				$( '#lang_locale' ).val( value[1] );
-				$( 'input[name="rtl"]' ).val( [value[2]] );
-				$( '#lang_name' ).val( selected[0] );
+		// Callback when selectmenu widget change event is triggered.
+		var changeCallback = function( event, ui ) {
+			var value = $( event.target ).val().split( ':' );
+			var selected = $( "option:selected", event.target ).text().split( ' - ' );
+			$( '#lang_slug' ).val( value[0] );
+			$( '#lang_locale' ).val( value[1] );
+			$( 'input[name="rtl"]' ).val( [value[2]] );
+			$( '#lang_name' ).val( selected[0] );
```

Comment:
Flag from the flags list dropdown is also a language field.
So I also put its value assignment in this new function even if we have to test the flags list dropdown presence.

To avoid to test twice if the flags list dropdown exists I also included the jQuery UI selectmenu flags list widget refresh in this function.